Tests/MapboxSearchIntegrationTests/OfflineIntegrationTests.swift

```diff
@@ -57,6 +57,18 @@ class OfflineIntegrationTests: MockServerIntegrationTestCase<GeocodingMockRespon
     func testLoadData() throws {
         clearData()
 
+        // Set up index observer before the fetch starts to validate changes after it completes
+        let indexChangedExpectation = expectation(description: "Received offline index changed event")
+        let offlineIndexObserver = OfflineIndexObserver(onIndexChangedBlock: { changeEvent in
+            _Logger.searchSDK.info("Index changed: \(changeEvent)")
+            indexChangedExpectation.fulfill()
+        }, onErrorBlock: { error in
+            _Logger.searchSDK.error("Encountered error in OfflineIndexObserver \(error)")
+            XCTFail(error.debugDescription)
```

Comment:
This fail occurs with dataset `test-dataset` (what the branch uses now) until this can be fixed in other PRs.